packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -217,7 +226,7 @@ export class AmplifyGraphqlApi extends Construct {
     this.codegenAssets = new CodegenAssets(this, 'AmplifyCodegenAssets', { modelSchema: definition.schema });
 
     this.resources = getGeneratedResources(this);
-    this.conflictResolution = conflictResolution;
+    this.dataStoreConfiguration = dataStoreConflictResolution;
```

Comment:
Now that we're doing some magic with this, I'd prefer to move this higher up in the constructor--like around L164. You could then bypass the whole `dataStoreConflictResolution` intermediate variable entirely and just operate on `this.dataStoreConfiguration` throughout the rest of the ctor.